base_dav/radicale/collection.py

```diff
@@ -120,14 +121,58 @@ def get(self, href):
                         record.write_date
                     ),
                 )
-        record = collection_model.browse(int(components[-1]))
+
+        try:
+            record = collection_model.browse(int(components[-1]))
+        except ValueError:
+            return None
         return Item(
             self,
             item=self.collection.to_vobject(record),
             href=href,
             last_modified=self._odoo_to_http_datetime(record.write_date),
         )
 
+    def upload(self, href, vobject_item):
+        components = self._split_path(href)
+
+        collection_model = self.env[self.collection.model_id.model]
+        if self.collection.dav_type == 'files':
+            return super().upload(href, vobject_item)
+
+        data = self.collection.from_vobject(vobject_item)
+
+        try:
+            record_id = int(components[-1])
+        except ValueError:
+            record_id = None
+
+        if not record_id:
+            record = collection_model.create(data)
+        else:
+            record = collection_model.browse(record_id)
+            record.write(data)
+
+        return Item(
+            self,
+            item=self.collection.to_vobject(record),
+            href=href,
```

Comment:
don't you need to adapt `href` here for the create case?

Reply:
You are right that might be a problem if the client doesn't always use list.